src/main/java/net/imglib2/img/basictypeaccess/Accesses.java

```diff
@@ -0,0 +1,451 @@
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package net.imglib2.img.basictypeaccess;
+
+/**
+ * Utility and helper methods for accesses ({@link ByteAccess} etc).
+ *
+ * @author Philipp Hanslovsky
+ */
+public interface Accesses
+{
+
+	/**
+	 *
+	 * Following {@link System#arraycopy}, copies {@code length} elements from
+	 * the specified source access, beginning at the specified position
+	 * {@code srcPos}, to the specified position of the destination array
+	 * {@code destPos}. A subsequence of access components are copied from the
+	 * source access referenced by {@code src} to the destination access
+	 * referenced by {@code dest}. The number of components copied is equal to
+	 * the {@code length} argument. The components at positions {@code srcPos}
+	 * through {@code srcPos+length-1} in the source array are copied into
+	 * positions {@code destPos} through {@code destPos+length-1}, respectively,
+	 * of the destination array.
+	 *
+	 * If the {@code src} and {@code dest} arguments refer to the same array
+	 * object, then the copying is performed as if the components at positions
+	 * {@code srcPos} through {@code srcPos+length-1} were first copied to a
+	 * temporary array with {@code length} components and then the contents of
+	 * the temporary array were copied into positions destPos through
+	 * {@code destPos+length-1} of the destination array.
+	 *
+	 * @param src
+	 *            the source access.
+	 * @param srcPos
+	 *            starting position in the source access.
+	 * @param dest
+	 *            the destination access.
+	 * @param destPos
+	 *            starting position in the destination access.
+	 * @param length
+	 *            the number of access elements to be copied
+	 */
+	public static void copy(
+			final ByteAccess src,
+			final int srcPos,
+			final ByteAccess dest,
+			final int destPos,
+			final int length )
+	{
+
+		if ( src == dest )
+		{
+			if ( srcPos == destPos ) { return; }
+			if ( srcPos < destPos )
```

Comment:
This looks incomplete. What if `destPos > srcPos`

Reply:
Addressed in e4897a5